### PR TITLE
React translations

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,8 +20,9 @@
     "async": "0.9.0",
     "autosize": "^1.18.13",
     "bootstrap": "3.1.1",
-    "i18next": "3.4.4",
-    "i18next-xhr-backend": "1.1.0",
+    "i18next": "7.0.0",
+    "i18next-browser-languagedetector": "*",
```
Comment (saghul) [inline package.json]: Please specify a version, we have been bitten by * recently.

File: react/features/app/components/AbstractApp.js [hunk 2]
```diff
@@ -134,11 +137,13 @@ export class AbstractApp extends Component {
 
         if (route) {
             return (
-                <Provider store = { this._getStore() }>
-                    {
-                        this._createElement(route.component)
-                    }
-                </Provider>
+                <I18nextProvider i18n = { i18n }>
```
Comment (saghul) [inline react/features/app/components/AbstractApp.js]: I think this can be simplified by passing the i18n prop to the `Provider` below, since `I18nextProvider` looks like a simple wrapper.

File: react/features/base/translation/Translation.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    ns: {
+        namespaces: [ 'main', 'languages' ],
+        defaultNs: 'main'
+    },
+    lngWhitelist: languages.getLanguages(),
+    fallbackOnNull: true,
+    fallbackOnEmpty: true,
+    useDataAttrOptions: true,
+    app: typeof interfaceConfig === 'undefined'
+        ? 'Jitsi Meet' : interfaceConfig.APP_NAME
```
Comment (saghul) [inline react/features/base/translation/Translation.js]: do we support null or '' as the app name? How about `interfaceConfig.APP_NAME || 'Jitsi Meet'`

**Comment by damencho** [comment]
React native complains about interfaceConfig not defined. Well null and '' is ok, we just cannot use the processor that replaces app name in some strings, we don't use it for now in native ... but we may need it at some point.

File: package.json (hunk 1 above)
Comment (damencho) [inline package.json]: I'll change the function to i18n, as we already have translate. And its easier for short one inside html in react, as strings are long enough.

File: react/features/app/components/AbstractApp.js (hunk 2 above)
Comment (damencho) [inline react/features/app/components/AbstractApp.js]: Well, ok, but this is how react-i18next should be implemented following their docs, what if in a newer version that provider changes. Yes we can add it then, but I'm just in a doubt about this, have no problem changing it.

File: package.json (hunk 1 above)
Comment (damencho) [inline package.json]: Hum i18n is not possible cause it clashes with variable in Translate Component inside react-i18next.

**Comment by damencho** [comment]
I tried replacing t function with one with custom name i18n but it is not possible cause this variable is already used in react-i18next. As we are familiar with function t from i18next lets stick with default one for now.

**Comment by damencho** [comment]
I removed some other temporary translation workarounds and also the version in package.json, I think I fixed all comments.

File: package.json [hunk 4]
```diff
@@ -34,10 +35,12 @@
     "postis": "^2.2.0",
     "react": "15.4.2",
     "react-dom": "15.4.2",
+    "react-i18next": "2.2.0",
     "react-native": "0.41.2",
     "react-native-background-timer": "1.0.0",
     "react-native-immersive": "0.0.4",
     "react-native-keep-awake": "^2.0.2",
+    "react-native-locale-detector": "1.0.1 ",
```
Comment (saghul) [inline package.json]: trailing space?

File: react/features/app/components/AbstractApp.js (hunk 2 above)
Comment (saghul) [inline react/features/app/components/AbstractApp.js]: Ok, let's leave it as is then.

File: react/features/base/translation/ConfigLanguageDetector.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+     * @returns {string} The default language if any.
+     */
+    lookup() {
+        return config.defaultLanguage;
+    },
+
+    /**
+     * Doesn't support caching.
+     */
+    cacheUserLanguage: Function.prototype
```
Comment (saghul) [inline react/features/base/translation/ConfigLanguageDetector.js]: is this supposed to be a noop function? In that case, how about `() => {}`, I find it easier to understand.

File: react/features/base/translation/LanguageDetector.native.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import locale from 'react-native-locale-detector';
+
+/**
+ * A language detector that uses native locale.
+ */
+export default {
+    init: Function.prototype,
```
Comment (saghul) [inline react/features/base/translation/LanguageDetector.native.js]: ditto

File: react/features/base/translation/LanguageDetector.native.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    type: 'languageDetector',
+    detect: () => locale,
+    cacheUserLanguage: Function.prototype
```
Comment (saghul) [inline react/features/base/translation/LanguageDetector.native.js]: ditto

File: react/features/base/translation/Translation.js (hunk 3 above)
Comment (saghul) [inline react/features/base/translation/Translation.js]: bump

File: modules/settings/Settings.js [hunk 8]
```diff
@@ -118,7 +118,6 @@ export default {
     },
     setLanguage: function (lang) {
         language = lang;
-        jitsiLocalStorage.setItem("language", lang);
```
Comment (saghul) [inline modules/settings/Settings.js]: Why is this removed? I guess because the local storage backend takes care, but then why do we load it from there on line 27?
